SWDEV-556212 - Update changelog for HIP 7.1 (#1261)
ROCm/rocm-systems: projects/clr/CHANGELOG.md

    - `hipGetDriverEntryPoint ` gets function pointer of a HIP API.
    - `hipSetValidDevices`      sets a default list of devices that can be used by HIP
    - `hipStreamGetId`          queries the id of a stream
* New HIP flags

>     - `hipMemLocationTypeHost`, allowing to extend the capability of handling the virtual memory management in host memory location, in addition to device memory.

Reviewer: ```suggestion
    - `hipMemLocationTypeHost`, enables handling virtual memory management in host memory location, in addition to device memory.
```

Author: Thanks. Updated.

    - `hipHostRegisterIoMemory` is supported in `hipHostRegister`, used to register I/O memory with HIP
    runtime so it can be accessed by the GPU.

> * Support for nested tile partitioning within cooperative groups, matching corresponding CUDA's functionality.

Reviewer: ```suggestion
* Support for nested tile partitioning within cooperative groups, matching NVIDIA CUDA functionality.
```

Author: Thanks, updated.

    - `hipGetDriverEntryPoint ` gets function pointer of a HIP API.
    - `hipSetValidDevices`      sets a default list of devices that can be used by HIP
    - `hipStreamGetId`          queries the id of a stream
* New HIP flags
    - `hipMemLocationTypeHost`, allowing to extend the capability of handling the virtual memory management in host memory location, in addition to device memory.

>     - `hipHostRegisterIoMemory` is supported in `hipHostRegister`.

Reviewer 2: what does it do?

Author: Added. Thanks.

* Support for nested tile partitioning within cooperative groups, matching corresponding CUDA's functionality.

### Resolved issues

* A segmentation fault occurred in application when capturing the same HIP graph from multiple streams with cross-stream dependencies.  HIP runtime fixed an issue where a forked stream joined to a parent stream which was not originally created with the API `hipStreamBeginCapture`.
* Different behavior of en-queuing command on a legacy stream during stream capture on AMD ROCM platform, compared with NVIDIA CUDA. HIP runtime now returns an error in this specific situation, to behave the same as CUDA.
* Failure of memory access fault occurred in rocm-examples test suite. When Heterogeneous Memory Management (HMM) is not supported in the driver, `hipMallocManaged` will only allocate system memory in HIP runtime.

### Optimized

* Improved hip module loading latency.
* Optimized kernel metadata retrieval during module post load.
* Optimized doorbell ring in HIP runtime, advantages the following for performance improvement,

>     - Makes efficient packet batching for HIP graphic launch,

Reviewer 2: graph, not graphic

Author: Thanks. Updated.

>     - Dynamically copies packets based on the max threshold if defined, otherwise staggers copy with power of 2.

Reviewer 2: what does this mean? and what's the threshold?

Author: Updated. Thanks.

* Reduced usage of the lock scope in events and kernel handling.
    - Switches to `shared_mutex` for event validation, uses `std::unique_lock` in HIP runtime to create/destroy event, instead of `scopedLock`.

>     - Reduces the `scopedLock` in handling of kernel execution. HcorrespondingIP runtime now calls `scopedLock` during kernel binary creation/initialization, doesn't call it again during kernel vector iteration before launch.

Reviewer 2: HIP, not HcorrespondingIP

Author: Corrected. Thanks.
